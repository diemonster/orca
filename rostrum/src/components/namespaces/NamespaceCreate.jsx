import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { namespaceCreateChangeInput, namespaceCreate } from '../../actions/namespaces';

class NamespaceCreate extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    event.preventDefault();

    const { dispatchNamespaceCreateChangeInput } = this.props;

    dispatchNamespaceCreateChangeInput(event.target.value);
  }

  handleSubmit(event) {
    event.preventDefault();

    const { dispatchNamespaceCreate, namespaceCreateInput } = this.props;

    dispatchNamespaceCreate(namespaceCreateInput);
  }

  render() {
    const { namespaceCreateInput } = this.props;

    return (
      <div className="create-namespace">
        <h2>Create Namespace</h2>
        <form className="namespace-input-form" onSubmit={this.handleSubmit}>
          <label className="namespace-label" htmlFor="namespace">
            New Namespace:
            <input
              type="text"
              value={namespaceCreateInput}
              onChange={this.handleChange}
              placeholder="enter namespace"
              id="namespace"
            />
            <button className="button" type="submit" value="submit">
              Create
            </button>
          </label>
        </form>
      </div>
    );
  }
}

NamespaceCreate.propTypes = {
  dispatchNamespaceCreate: PropTypes.func.isRequired,
  dispatchNamespaceCreateChangeInput: PropTypes.func.isRequired,
  namespaceCreateInput: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  namespaceCreateInput: state.namespace.namespaceCreateInput,
});

const mapDispatchToProps = dispatch => ({
  dispatchNamespaceCreateChangeInput: namespaceCreateInput => (
    dispatch(namespaceCreateChangeInput(namespaceCreateInput))
  ),
  dispatchNamespaceCreate: name => dispatch(namespaceCreate(name)),
});

export default connect(mapStateToProps, mapDispatchToProps)(NamespaceCreate);
