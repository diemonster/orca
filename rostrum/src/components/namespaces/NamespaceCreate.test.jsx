import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { NamespaceCreate } from './NamespaceCreate';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('NamespaceCreate', () => {
    it('should render itself and its subcomponents', () => {
      const props = {
        dispatchNamespaceCreate: jest.fn(),
        dispatchNamespaceCreateChangeInput: jest.fn(),
        namespaceCreateInput: '',
      };

      const namespaceCreate = shallow(<NamespaceCreate {...props} />);

      expect(namespaceCreate).toMatchSnapshot();
    });

    it('appropriately calls dispatchNamespaceCreateChangeInput() on input change', () => {
      const mockFn = jest.fn();
      const props = {
        dispatchNamespaceCreate: jest.fn(),
        dispatchNamespaceCreateChangeInput: mockFn,
        namespaceCreateInput: '',
      };

      const namespaceCreate = shallow(<NamespaceCreate {...props} />);

      const event = { preventDefault: jest.fn(), target: { value: 'a' } };
      namespaceCreate.find('input').simulate('change', event);
      expect(mockFn).toBeCalledWith('a');
    });

    it('appropriately calls dispatchNamespaceCreate() on form submit', () => {
      const mockFn = jest.fn();
      const props = {
        dispatchNamespaceCreate: mockFn,
        dispatchNamespaceCreateChangeInput: jest.fn(),
        namespaceCreateInput: 'new-namespace',
      };

      const namespaceCreate = shallow(<NamespaceCreate {...props} />);

      const event = { preventDefault: jest.fn(), target: { value: 'new-namespace' } };
      namespaceCreate.find('form').simulate('submit', event);
      expect(mockFn).toBeCalledWith('new-namespace');
    });
  });
});
