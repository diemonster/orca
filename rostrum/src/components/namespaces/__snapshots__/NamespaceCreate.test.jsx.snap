// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NamespaceCreate should render itself and its subcomponents 1`] = `
<div
  className="create-namespace"
>
  <h2>
    Create Namespace
  </h2>
  <form
    className="namespace-input-form"
    id="namespace-create-input-form"
    onSubmit={[Function]}
  >
    <label
      className="namespace-input-label"
      htmlFor="namespace-create-input"
    >
      New Namespace:
    </label>
    <input
      id="namespace-create-input"
      name="namespace-create-input"
      onChange={[Function]}
      placeholder="enter namespace"
      type="text"
      value=""
    />
    <button
      className="button"
      type="submit"
      value="submit"
    >
      Create
    </button>
  </form>
</div>
`;
